spec/models/new_msa_component_event_spec.rb

```diff
@@ -31,4 +32,18 @@
       expect(event.errors[:environment]).to eql [t('components.errors.invalid_environment')]
     end
   end
-end
+
+  context 'entity_id' do
+    it 'must be provided' do
+      event = build(:new_msa_component_event, entity_id: '')
+      expect(event).to_not be_valid
+      expect(event.errors[:entity_id]).to eql [t('components.errors.missing_entity_id')]
+    end
+
+    it 'must not contain spaces' do
+      event = build(:new_msa_component_event, entity_id: ' test entity id ')
```

Comment:
This is mixing two checks in one

This should be split into two tests:
- must not contain spaces - provide test entity id 'test entity id' (this test)
- strips whitespace - provide test entity id ` test-entity-id ` and check the created component has no spaces in entity id